container creation/get-deployment-details.sh

```diff
@@ -0,0 +1,269 @@
+#!/bin/bash
+# Helper script to gather project details for automatic deployment
+# Modified July 17th, 2025 by Maxwell Klema
+# ------------------------------------------
+
+# Define color variables (works on both light and dark backgrounds)
+RESET="\033[0m"
+BOLD="\033[1m"
+MAGENTA='\033[35m'
+
+echo -e "${BOLD}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${RESET}"
+echo -e "${BOLD}${MAGENTA}🌐 Let's Get Your Project Automatically Deployed ${RESET}"
+echo -e "${BOLD}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${RESET}"
+
+# Get and validate project repository ========
+
+if [ -z "$PROJECT_REPOSITORY" ]; then
+    read -p "🚀 Paste the link to your project repository →  " PROJECT_REPOSITORY
+fi
+
+CheckRepository() {
+    PROJECT_REPOSITORY_SHORTENED=${PROJECT_REPOSITORY#*github.com/}
+    PROJECT_REPOSITORY_SHORTENED=${PROJECT_REPOSITORY_SHORTENED%.git}
+    REPOSITORY_EXISTS=$(curl -s -o /dev/null -w "%{http_code}" https://github.com/$RROJECT_REPOSITORY)
+}
+
+CheckRepository
+
+while [ "$REPOSITORY_EXISTS" != "200" ]; do
+    echo "⚠️ The repository link you provided, \"$PROJECT_REPOSITORY\" was not valid."
+    read -p "🚀 Paste the link to your project repository →  " PROJECT_REPOSITORY
+    CheckRepository
+done
+
+# Get Repository Branch ========
+
+if [ -z "$PROJECT_BRANCH" ]; then
+    read -p "🪾  Enter the project branch to deploy from (leave blank for \"main\") → " PROJECT_BRANCH
+fi
+
+if [ "$PROJECT_BRANCH" == "" ]; then
+    PROJECT_BRANCH="main"
+fi
+
+REPOSITORY_BRANCH_EXISTS=$(curl -s -o /dev/null -w "%{http_code}" $PROJECT_REPOSITORY/tree/$PROJECT_BRANCH)
+while [ "$REPOSITORY_BRANCH_EXISTS" != "200" ]; do
+    echo "⚠️ The branch you provided, \"$PROJECT_BRANCH\", does not exist on repository at \"$PROJECT_REPOSITORY\"."
+    read -p "🪾  Enter the project branch to deploy from (leave blank for \"main\") → " PROJECT_BRANCH
+    if [ "PROJECT_BRANCH" == "" ]; then
```

Comment:
This original conditional looks to be making a comparison with a a literal string, perhaps for better accuracy we can check the actual _value_ of the `$PROJECT_BRANCH` var.
```suggestion
   if [ -z "$PROJECT_BRANCH" ]; then
```

Reply:
I agree. I will re-factor these comparisons with empty strings to use the empty flag, -z.